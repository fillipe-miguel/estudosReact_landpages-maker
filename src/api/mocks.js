/* eslint-disable import/prefer-default-export */
export const pageDataMock = [
  {
    title: 'Title teste 01',
    slug: 'title-teste-01',
    menu: {
      logo_link: '#link-teste',
      logo_text: 'LOGO TESTE',
      logo: {
        data: {
          attributes: {
            url: 'https://url-da-imagem_teste.svg',
          },
        },
      },
      menu_link: [
        {
          link_text: 'Text do link test',
          url: '#link-test',
          open_in_new_tab: true,
        },
      ],
    },
    sections: [
      // ================================= Layout grid column / section two columns =================================
      {
        __component: 'section.section-two-columns',
        title: 'Essa é a seção de duas colunas',
        description: 'Descrição da seção de duas colunas',
        image: {
          data: {
            attributes: {
              url: 'https://layoutgridcolumn-test.jpg',
            },
          },
        },
        metadata: {
          section_id: 'section-two-columns',
          background: false,
        },
      },

      // ================================= Layout center content / section content ==================================
      {
        __component: 'section.section-content',
        title: 'Essa é a seção de uma coluna e só texto',
        content: 'lorem lorem lorem lorem lorem',
        metadata: {
          section_id: 'section-center-content',
          background: true,
        },
      },

      // ================================= Layout grid describe/ section grid text =================================
      {
        __component: 'section.section-grid',
        title: 'Aqui é a seção de grid com texto',
        description: 'Vários textos ai olha que loucura!',
        metadata: {
          section_id: 'grid-text',
          background: true,
        },
        text_grid: [
          {
            title: 'Teste 1',
            description: 'Lorem Ipsum teste',
          },
          {
            title: 'Teste 2 ',
            description:
              "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
          },
          {
            title: 'Teste 3',
            description:
              "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
          },
        ],
        image_grid: [],
      },

      // ================================= Section grid image / Layout grid gallery =================================
      {
        __component: 'section.section-grid',
        title: 'Section grid com imagens / gallery',
        description: 'Éssa é a seção com o grid de imagens muito loucas',
        metadata: {
          name: 'section-grid-gallery',
          section_id: 'section-grid-gallery',
          background: true,
        },
        text_grid: [],
        image_grid: [
          {
            image: {
              data: {
                attributes: {
                  alternativeText: 'Mulher segurando o peso',
                  url: 'https://section-grid-gallery.jpg',
                },
              },
            },
          },
        ],
      },
    ],
    footer: '<p>Teste aqui</p>',
  },
];
