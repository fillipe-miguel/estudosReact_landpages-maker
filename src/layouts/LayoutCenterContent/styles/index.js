import styled, { css } from 'styled-components';

export const LayoutCenterContentRoot = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const InnerHtml = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.large};
    text-align: justify;

    & h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      text-align: center;
    }
  `}
`;

export default LayoutCenterContentRoot;
