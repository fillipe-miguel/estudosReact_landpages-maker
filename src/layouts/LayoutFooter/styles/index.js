import styled, { css } from 'styled-components';

export const LayoutFooterRoot = styled.footer`
  ${({ theme }) => css`
    height: 8rem;
    text-align: center;
    font-size: ${theme.fonts.sizes.xsmall};
    background-color: ${theme.colors.white};

    a {
      pointer-events: all !important;
      text-decoration: underline !important;
    }
  `}
`;

export default LayoutFooterRoot;
