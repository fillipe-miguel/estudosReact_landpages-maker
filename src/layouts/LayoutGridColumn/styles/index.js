import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';

export const LayoutGridColumnRoot = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;

    gap: ${theme.spacings.xxlarge};

    & ${HeadingStyles.HeadingRoot} {
      margin-bottom: ${theme.spacings.small};
    }
  `}

  // ================================= Medias =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      flex-direction: column;

      & ${HeadingStyles.HeadingRoot} {
        margin-bottom: ${theme.spacings.xlarge};
      }
    }
  `}
`;

export const TextContainer = styled.div`
  ${() => css`
    max-width: 35rem;
  `}

  // ================================= Medias =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      min-width: auto;
      text-align: center;
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css`
    max-width: 64rem;
  `}
`;

export const Image = styled.img`
  ${() => css`
    max-width: 100%;
  `}
`;
