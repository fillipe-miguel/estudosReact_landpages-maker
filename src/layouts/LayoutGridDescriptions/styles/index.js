import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xxlarge};
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    & ${HeadingStyles.HeadingRoot} {
      position: relative;
      left: 5rem;
      margin: ${theme.spacings.medium} 0;
    }

    & ${HeadingStyles.HeadingRoot}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xxxlarge};
      left: -5rem;
      top: -2.5rem;
      transform: skew(-15deg);
    }
  `}
`;
