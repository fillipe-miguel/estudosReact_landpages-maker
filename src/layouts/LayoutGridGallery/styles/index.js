import styled, { css } from 'styled-components';

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xxlarge};
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(270px, 1fr));
    gap: ${theme.spacings.medium};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
    height: fit-content;

    &:hover > img {
      transition: all 0.3s ease-in-out;
      transform: scale(1.2) rotate(9deg);
    }

    img {
      pointer-events: none;
      max-width: 100%;
    }
  `}
`;
