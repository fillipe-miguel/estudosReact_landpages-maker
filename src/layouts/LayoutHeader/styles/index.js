import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';
import { SectionContainerStyles } from '../../../sections';

export const LayoutHeaderRoot = styled.header`
  position: fixed;
  z-index: 5;
  top: 0;
  right: 0;
  left: 0;

  & > ${SectionContainerStyles.SectionContainerRoot} {
    padding-top: ${({theme}) => theme.spacings.small};
    padding-bottom: ${({theme}) => theme.spacings.small};;
  }

  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    border-bottom: 1px solid ${theme.colors.mediumGray};
  `}

  // ================================= MEDIAS  =================================
  ${({ theme, ...props }) => css`
    @media ${theme.medias.lteMedium} {
      bottom: 0;
      height: 100vh;
      visibility: ${props.visible ? 'visibility' : 'hidden'};
      opacity: ${props.visible ? '1' : '0'};
      transition: all 0.3s ease-in-out;

      & > ${SectionContainerStyles.SectionContainerRoot} {
        display: grid;
        align-items: center;

        height: 100%;
        margin: 0;

        overflow-y: auto;
      }

      & ${HeadingStyles.HeadingRoot} {
        padding-bottom: ${theme.spacings.large};
      }
    }
  `}
`;

export const HeaderContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  // ================================= MEDIAS  =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      display: flex;
      flex-direction: column;
      text-align: center;
      padding: ${theme.spacings.xxlarge};
    }
  `}
`;

export const ToggleMenuButton = styled.button`
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;

  display: none;

  font-size: 2.5rem;
  padding: 0.5rem;

  cursor: pointer;
  border: none;

  ${({ theme }) => css`
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
  `}

  // ================================= MEDIAS  =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      display: flex;
      align-items: center;
    }
  `}
`;

export default LayoutHeaderRoot;
