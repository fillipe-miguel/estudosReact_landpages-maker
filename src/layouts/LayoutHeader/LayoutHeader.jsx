import P from 'prop-types';
import React, { useState } from 'react';
import { MdOutlineClose, MdOutlineMenu } from 'react-icons/md';

import { LogoLink, MenuNav } from '../../components';
import { SectionContainer } from '../../sections';

import * as Styled from './styles';

function LayoutHeader({ links = [], logoData }) {
  const [visible, setVisible] = useState(false);

  const toggleVisibility = () => setVisible((current) => !current);

  return (
    <>
      <Styled.ToggleMenuButton
        onClick={() => {
          toggleVisibility();
        }}
        title={visible ? 'Close menu' : 'Open menu'}
        aria-label="Open/Close Menu"
      >
        {visible ? (
          <MdOutlineClose aria-label="Close menu icon" />
        ) : (
          <MdOutlineMenu aria-label="Open menu icon" />
        )}
      </Styled.ToggleMenuButton>

      <Styled.LayoutHeaderRoot
        visible={visible}
        onClick={() => {
          toggleVisibility();
        }}
      >
        <SectionContainer>
          <Styled.HeaderContainer>
            <LogoLink {...logoData} />
            <MenuNav links={links} />
          </Styled.HeaderContainer>
        </SectionContainer>
      </Styled.LayoutHeaderRoot>
    </>
  );
}

LayoutHeader.defaultProps = {
  // Aqui as propriedades default
};

LayoutHeader.propTypes = {
  ...MenuNav.propTypes,
  logoData: P.shape(LogoLink.propTypes).isRequired,
};

export default LayoutHeader;
