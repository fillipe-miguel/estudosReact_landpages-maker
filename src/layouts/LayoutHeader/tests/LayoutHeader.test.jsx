/* eslint-disable no-undef */
import { fireEvent, render, screen } from '@testing-library/react';

import LayoutHeader from '..';

import * as mocks from '../../../mocks';
import { MyThemeProvider, theme } from '../../../styles';

describe('<LayoutHeader />', () => {
  // ================================= Teste 01 =================================
  it('Should render LogoLink and MenuNav', () => {
    render(
      <MyThemeProvider>
        <LayoutHeader
          links={mocks.components.links}
          logoData={mocks.components.logoData}
        />
      </MyThemeProvider>,
    );

    // ! Testando o se o LogoLink se encontra
    expect(
      screen.queryByRole('heading', { name: mocks.components.logoData.text }),
    ).toBeInTheDocument();

    // ! Testando o se o MenuNav se encontra
    expect(
      screen.queryByRole('navigation', { name: 'Main menu' }),
    ).toBeInTheDocument();
  });

  // ================================= Teste 02 =================================
  it('Should render mobile Menu and the button Open/Close', () => {
    render(
      <MyThemeProvider>
        <LayoutHeader
          links={mocks.components.links}
          logoData={mocks.components.logoData}
        />
      </MyThemeProvider>,
    );

    const button = screen.getByLabelText('Open/Close Menu');
    const menuContainer = button.nextSibling;

    // ! Testando se o botão NÃO ta na tela quando PC
    expect(button).toHaveStyleRule('display', 'none');

    // ! Testando se o botão ta na tela quando mobile
    expect(button).toHaveStyleRule('display', 'flex', {
      media: theme.medias.lteMedium,
    });

    // ! Testando se o menu ta oculto
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.medias.lteMedium,
    });

    // ! Testando se o botão está com o formato de abrir
    expect(screen.getByLabelText('Open menu icon')).toBeInTheDocument();
  });

  // ================================= Teste 03 =================================
  it('Should button Open/Close menu button work', () => {
    render(
      <MyThemeProvider>
        <LayoutHeader
          links={mocks.components.links}
          logoData={mocks.components.logoData}
        />
      </MyThemeProvider>,
    );

    const button = screen.getByLabelText('Open/Close Menu');
    const menuContainer = button.nextSibling;

    // * Aqui o botão foi clicado para abrir !!!

    fireEvent.click(button); // Clicando no botão

    // ! Testando se o menu ta visível
    expect(menuContainer).toHaveStyleRule('opacity', '1', {
      media: theme.medias.lteMedium,
    });

    // ! Testando se o botão está com o formato de fechar
    expect(screen.getByLabelText('Close menu icon')).toBeInTheDocument();

    // * Aqui o botão foi clicado para fechar !!!

    fireEvent.click(button); // Clicando no botão

    // ! Testando se o menu ta visível
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.medias.lteMedium,
    });

    // ! Testando se o botão está com o formato de fechar
    expect(screen.getByLabelText('Open menu icon')).toBeInTheDocument();
  });

  /// ================================= Teste 04 =================================
  it('Should close menu clicking on container', () => {
    render(
      <MyThemeProvider>
        <LayoutHeader
          links={mocks.components.links}
          logoData={mocks.components.logoData}
        />
      </MyThemeProvider>,
    );

    const button = screen.getByLabelText('Open/Close Menu');
    const menuContainer = button.nextSibling;

    fireEvent.click(menuContainer); // Clicando no botão para abrir

    // * Aqui o menuContainer clicado !!!

    fireEvent.click(menuContainer); // Clicando no Container para fechar

    // ! Testando se o menu ta visível
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.medias.lteMedium,
    });

    // ! Testando se o botão está com o formato de fechar
    expect(screen.getByLabelText('Open menu icon')).toBeInTheDocument();
  });

  // ================================= Teste 05 =================================
  it('Should not render links', () => {
    render(
      <MyThemeProvider>
        <LayoutHeader logoData={mocks.components.logoData} />
      </MyThemeProvider>,
    );

    // ! Vendo se o MenuNav tem algum link filho
    expect(
      screen.queryByRole('navigation', { name: 'Main menu' }).firstChild,
    ).not.toBeInTheDocument();
  });

  // ================================= Snapshot =================================
  it('Should not render links', () => {
    const { container } = render(
      <MyThemeProvider>
        <LayoutHeader
          logoData={mocks.components.logoData}
          links={mocks.components.links}
        />
      </MyThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
