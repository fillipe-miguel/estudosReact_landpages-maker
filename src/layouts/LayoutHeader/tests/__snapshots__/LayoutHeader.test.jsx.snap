// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutHeader /> Should not render links 1`] = `
.c7 {
  color: #0A1128;
  text-transform: uppercase;
  font-size: 2.4rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > img {
  max-height: 3rem;
}

.c9 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
}

.c9::after {
  content: '';
  position: absolute;
  bottom: 1rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #DC143C;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c9:hover::after {
  left: 25%;
  width: 50%;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c3 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c1 {
  position: fixed;
  z-index: 5;
  top: 0;
  right: 0;
  left: 0;
  background-color: #FFF;
  border-bottom: 1px solid #DDDDDD;
}

.c1 > .c2 {
  padding-top: 1.6rem;
  padding-bottom: 1.6rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  display: none;
  font-size: 2.5rem;
  padding: 0.5rem;
  cursor: pointer;
  border: none;
  background-color: #0A1128;
  color: #FFF;
}

@media (max-width:768px) {
  .c7 {
    font-size: 2.4rem;
  }
}

@media (max-width:768px) {
  .c8 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:768px) {
  .c3 {
    padding: 1.6rem;
  }
}

@media (max-width:768px) {
  .c1 {
    bottom: 0;
    height: 100vh;
    visibility: hidden;
    opacity: 0;
    -webkit-transition: all 0.3s ease-in-out;
    transition: all 0.3s ease-in-out;
  }

  .c1 > .c2 {
    display: grid;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    height: 100%;
    margin: 0;
    overflow-y: auto;
  }

  .c1 .c6 {
    padding-bottom: 3.2rem;
  }
}

@media (max-width:768px) {
  .c4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    text-align: center;
    padding: 4.8rem;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <button
    aria-label="Open/Close Menu"
    class="c0"
    title="Open menu"
  >
    <svg
      aria-label="Open menu icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <header
    class="c1"
  >
    <div
      class="c2 c3"
    >
      <div
        class="c4"
      >
        <a
          class="c5"
          href="#target"
        >
          <h2
            class="c6 c7"
          >
            logo link
          </h2>
        </a>
        <nav
          aria-label="Main menu"
          class="c8"
        >
          <a
            class="c9"
            href="https://www.google.com.br"
            target="_blank"
          >
            Google
          </a>
          <a
            class="c9"
            href="https://youtube.com"
            target="_blank"
          >
            YouTube
          </a>
          <a
            class="c9"
            href="https://github.com/fillipe-miguel"
            target="_self"
          >
            My GitHub
          </a>
          <a
            class="c9"
            href="https://www.tiktok.com/pt-BR/"
            target="_blank"
          >
            TikTok
          </a>
          <a
            class="c9"
            href="https://www.udemy.com/"
            target="_blank"
          >
            Udemy
          </a>
        </nav>
      </div>
    </div>
  </header>
</div>
`;
